fuse/fuse.cpp

```diff
@@ -167,22 +168,45 @@ int main(int argc, char *argv[])
 		if (stdin.empty()) stdin = "/dev/null";
 
 #ifdef __linux__
-		// Allow overriding this to; e.g /build
-		std::string workspace = cwd;
-
 		uid_t real_euid = geteuid();
 		gid_t real_egid = getegid();
 
+		// Allow overriding this to a repeatable build directory
+		std::string workspace = cwd;
+		uid_t euid = real_euid;
+		gid_t egid = real_egid;
+		int flags = CLONE_NEWNS|CLONE_NEWUSER;
+
+		for (auto &res : jast.get("resources").children) {
+			std::string &key = res.second.value;
+			if (key == "isolate/user") euid = egid = 0;
+			if (key == "isolate/host") flags |= CLONE_NEWUTS;
+			if (key == "isolate/net") flags |= CLONE_NEWNET;
+			if (key == "isolate/workspace") {
+				if (access("/var/cache/wake", R_OK) == 0) {
+					workspace = "/var/cache/wake";
+				} else {
+					workspace = exedir + "/../../build/wake";
```

Comment:
Maybe this should be configurable via environment variable? You could still have this as the final fallback, but it could go in the order of 1) env var, 2) /var/cache, 3) relative to exedir?

Reply:
I would rather not make this any more complicated than it already is.